app/src/main/java/com/codesoom/assignment/application/member/MemberCreator.java

```diff
@@ -0,0 +1,19 @@
+package com.codesoom.assignment.application.member;
+
+import com.codesoom.assignment.domain.Member;
+import com.codesoom.assignment.domain.MemberRepository;
+import com.codesoom.assignment.web.shop.member.dto.MemberCreateRequest;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class MemberCreator {
+    private final MemberRepository memberRepository;
+
+    public Member create(MemberCreateRequest memberCreateRequest) {
+        Member createdMember = memberCreateRequest.toMember();
+        Member result = memberRepository.save(createdMember);
+        return result;
```

Comment:
생성된 결과를 표현하기 위해서 `result`라는 이름으로 변수를 만드신 것 같습니다. 여기서는 변수할당을 하지 않아도 되는 이유가 몇가지 존재하는데 

1. 우변의 메서드가 이미 충분히 설명하고 있다고 볼 수 있습니다. repository.save의 결과는 Member일테니, 좌변에 변수가 필요없다고 볼 수 있습니다.
2. 변수를 할당한다는 것은 변수로 조작을 할거라고 보여질 수 있습니다. 다른 누군가가 result라는 변수가 있으니 무언가 조작할 수 있게 되죠. 만약 조작하지 못하도록 하고 싶다면 오히려 변수 할당을 하지 않아야 합니다.

따라서 다음과 같이 바로 반환해도 좋을 것 같습니다.

```suggestion
        return memberRepository.save(createdMember);
```

Reply:
아하 메소드에서 만들어진 객체를 변수선언을 하면 조작을 암시할 수 있겠군요.. 거기까진 생각을 못했던 것 같습니다!! 
참고하여 수정하겠습니다!